app/lib/boolean-phrase-parser.js

```diff
@@ -0,0 +1,208 @@
+(function (_) {
```

Comment:
the purpose of passing arguments to the IIFE is to allow the module not to be defendant on specific external library names. it improves look-up speed since the local scope is searched before moving on to the window. and it will help with the minification of the files later on. shouldn't we pass them all. am I missing something?